RealmSwift/SwiftUI.swift

```diff
@@ -142,130 +131,103 @@ extension RealmBinding: RandomAccessCollection, BidirectionalCollection, Collect
     }
     /// :nodoc:
     public var startIndex: Index {
-        return wrappedValue.startIndex
+        return get().startIndex
     }
     /// :nodoc:
     public var endIndex: Index {
-        return wrappedValue.endIndex
+        return get().endIndex
     }
     /// :nodoc:
     public func index(before i: Index) -> Index {
-        return wrappedValue.index(before: i)
+        return get().index(before: i)
     }
     /// :nodoc:
     public func index(after i: Index) -> Index {
-        return wrappedValue.index(after: i)
+        return get().index(after: i)
     }
     /// :nodoc:
     public var indices: Indices {
-        return wrappedValue.indices
+        return get().indices
     }
     /// :nodoc:
     public subscript(position: Index) -> Element {
         get {
-            self.wrappedValue[position]
+            if self.get().isFrozen { throwRealmException("Cannot get values from a frozen realm") }
+            return self.get()[position]
```

Comment:
get() can do some nontrivial work so this should avoid calling it twice in one function.